onnxscript/rewriter/llama_rule_sets.py

```diff
@@ -2,13 +2,86 @@
 # Licensed under the MIT License.
 from __future__ import annotations
 
+import numpy as np
+import onnx.numpy_helper
+
 import onnxscript.ir as ir
 import onnxscript.rewriter.no_op as no_op
 import onnxscript.rewriter.pattern as orp
 
 op = orp.onnxop
 
 
+class CastIdentity(orp.RewriteRuleAsClass):
+    """Replaces ``Cast(., to=to)`` by ``Identity`` if possible."""
```

Comment:
Is it possible to elide the node in the optimizer?

Reply:
Not sure. Using idenity avoids renaming everything and we can still remove all identity nodes after every iteration. It is very fast.